test/bpf_tests/bpf_test.go

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Currently the distinction seems to be that `/bpf` is for C code and the rest for Go. You can't put Go and C files in the same directory since Go will automatically attempt to compile .c files as CGO, so any Go code would have to live in a separate directory anyway. Having 1 Go file in among all the C code felt strange, but I agree that it would be better to co-locate code that works together, perhaps not just for the tests but other parts of cilium as well